### PR TITLE
Support new Gelato relay API

### PR DESCRIPTION
**Pull request by tkporter**
* Moves over to using the new Gelato relay API in pretty much the same capacity as before (i.e. just using it to unconditionally send message process txs to)
* The new API requires us to use an API key instead of signing txs with our signer. I moved some things around in the settings/config to avoid needing to specify the API key for each chain that Gelato is enabled for. Instead, now we just (optionally) specify the API key once as a part of the abacus-base `Settings`. Each ChainSetup now is configured with an enum determining how transaction submission should work. `Signer` is the default (i.e. just normal EOA signing & sending off these txs with eth_sendTransaction), and `Gelato` is the alternative. If a chain is configured to use Gelato, the abacus-base GelatoConf must be specified.
* Renamed forward requests to sponsored calls.
* Removed a bunch of code related to Gelato that was no longer relevant -- lots of params are gone now, there's no need for EIP-712 signing anymore, etc
* The Gelato API key must be created as a GCP secret called `${environment}-gelato-api-key` manually - however the infra script will ensure it's present and give an error if it's not

### CODE REVIEW

File: rust/gelato/src/sponsored_call.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+pub struct SponsoredCallArgs {
+    pub chain_id: Chain,
+    pub target: Address,
+    pub data: Bytes,
+
+    // U256 by default serializes as a 0x-prefixed hexadecimal string.
+    // Gelato's API expects the gasLimit to be a decimal string.
+    #[serde(serialize_with = "serialize_as_decimal_str")]
+    pub gas_limit: Option<U256>,
+    pub retries: Option<u32>,
```
Comment (mattiecnvr) [inline rust/gelato/src/sponsored_call.rs:18]: Is there a difference between `Some(0)` and `None`?

File: rust/gelato/src/sponsored_call.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+#[derive(Debug, Clone)]
+pub struct SponsoredCallCall<'a> {
```
Comment (mattiecnvr) [inline rust/gelato/src/sponsored_call.rs:22]: Uhh, `...CallCall`?
Can we get some brief docs at least on the intent of these structs.

File: rust/gelato/src/sponsored_call.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    pub http: reqwest::Client,
+    pub args: &'a SponsoredCallArgs,
+    pub sponsor_api_key: &'a str,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SponsoredCallCallResult {
```
Comment (mattiecnvr) [inline rust/gelato/src/sponsored_call.rs:29]: These names are just getting too funny

**Review by mattiecnvr — APPROVED** [review]

File: rust/gelato/src/sponsored_call.rs (hunk 1 above)
Comment (tkporter) [reply rust/gelato/src/sponsored_call.rs:18]: yeah the API will default to 5, adding this as a comment

File: rust/gelato/src/sponsored_call.rs (hunk 2 above)
Comment (tkporter) [reply rust/gelato/src/sponsored_call.rs:22]: lmao you're totally right. This is the result of a `s/ForwardRequest/SponsoredCall/g`. I'll rename to `SponsoredCallApiCall` which is still not perfect but at least slightly less horrible

**Pull request merged**